CONTRIBUTING.md

```diff
@@ -1,48 +1,103 @@
+    * Or if a runtime setting is not possible a preprocessor switch that allows creating firmware without the change.
+* Changes that massively increase image size (> 5% of total memory) also require a preprocessor switch starting with "FEATURE_" so they can be enabled or disabled.
+* If the Pull request requires changes in structure of user files (e.g. project/synth or other xml files) or flash configuration.
+    * It must ensure that files or flash configuration created with the official firmware or previous community releases stay compatible or are automatically upgraded (upward compatibility).
+    * If possible files and flash configuration created with the changes of the Pull request can be used with older firmwares not having the change (downward compatibility). Older firmwares must not break if a newer configuration was stored on the device.
+* If the Pull request changes end user behavoir or introduces new features a new entry in the [CommunityFeatures.md](CommunityFeatures.md) file needs to be created in the preexisting style describing the feature and it's options as a small manual to users. This includes all runtime and compile time flags which shall be named in respective sections.
 
-* Pull requests exclusively fixing insignificant typos in code comments, one-off formatting mistakes, and whitespace will be rejected (though these may be collected together in a bulk fix for a later date).
+## Workflow
 
-* Automated PR submissions by clout/karma-seeking accounts (see also: [Hacktoberfest](https://blog.domenic.me/hacktoberfest/)) will not be tolerated or accepted.
+Please follow the following steps for every pull request to ensure every contribution can be handled as frictionless and transparent as possible: 
 
-## Documentation
+1. Create a [Draft Pull request](https://github.blog/2019-02-14-introducing-draft-pull-requests/) including a description on what will be changed and what impact you expect. The title can start with "[Draft] " during development.
+2. Work on the Pull request
+3. Before a Pull request can be considered ready all upstream changes from the ```develop``` branch need to be merged into it. It is the duty of everyone to help make merging into ```develop``` as painless as possible so please try to align if you see that your Pull requests works in a similar area as another one.
+4. Once the Pull request is ready, fulfills all requirements outlined above and is up to date with the ```develop``` branch it can be converted from Draft and marked as ready for review.
+5. If possible every Pull request should have at least two reviews. Reviews from community members not mentioned in the [CODEOWNERS](CODEOWNERS) file should be taken serious and used as an important source of feedback but have no decisional power on what gets merged into the ```develop``` branch.
```

Comment:
At this stage in the project, I think 2 "decisional power" reviews might be onerous, but as long as it's a "nice to have" rather than a hard and fast requirement I'm okay with it. Might want to distinguish between reviews of c/c++/asm code and workflow/peripheral code as well.

Reply:
I will update the wording to make it more explicit that one is enough. 

As for distinction between "low level" and other code I'm open to add it but I have a hard time to come up with a wording that makes sense. I tend to say that changes in all areas can have unexpected consequences impacting the whole device.

Reply:
True enough.